src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -131,54 +133,12 @@ extension RiskProvider: RiskProviding {
 			}
 			#endif
 
-			self._requestRiskLevel(userInitiated: userInitiated, ignoreCachedSummary: ignoreCachedSummary, completion: completion)
-		}
-	}
-
-	private func determineSummary(
-		userInitiated: Bool,
-		ignoreCachedSummary: Bool = false,
-		appConfiguration: SAP_Internal_ApplicationConfiguration,
-		completion: @escaping (Result<SummaryMetadata, RiskCalculationError>) -> Void
-	) {
-		Log.info("RiskProvider: Determine summeries.", log: .riskDetection)
-
-		if !ignoreCachedSummary {
-			// Here we are in automatic mode and thus we have to check the validity of the current summary.
-			let enoughTimeHasPassed = riskProvidingConfiguration.shouldPerformExposureDetection(
-				activeTracingHours: store.tracingStatusHistory.activeTracing().inHours,
-				lastExposureDetectionDate: store.summary?.date
-			)
-
-			let shouldDetectExposures = (riskProvidingConfiguration.detectionMode == .manual && userInitiated) || riskProvidingConfiguration.detectionMode == .automatic
-
-			if !enoughTimeHasPassed || !self.exposureManagerState.isGood || !shouldDetectExposures {
-				if let summary = store.summary {
-					completion(.success(summary))
-				} else {
-					completion(.failure(.missingCachedSummary))
-				}
-				return
+			// Erase optionality.
+			let completion: Completion = { result in
+				completion?(result)
```

Comment:
What's happening here?

Reply:
Resolving optionality for the completion closure for the rest of the code.